torch/functional.py

```diff
@@ -102,21 +102,37 @@ def broadcast_shapes(*shapes):
     """
     # This wrapper exists to support variadic args.
     # TODO Movie this to C++ once the jit has better support for torch.Size.
-    with torch.no_grad():
-        scalar = torch.zeros((), device="cpu")
-
-        for shape in [*shapes]:
-            if isinstance(shape, int) and shape < 0:
-                raise RuntimeError(rf"Trying to create tensor with negative dimension {shape}: [{shape}]")
-            if isinstance(shape, tuple):
-                for ele in shape:
-                    if ele < 0:
-                        raise RuntimeError(rf"Trying to create tensor with negative dimension {ele}: [{ele}]")
-
-        tensors = [scalar.expand(shape) for shape in shapes]
-        tensors = broadcast_tensors(*tensors)
-        return tensors[0].shape
-
+    for shape in shapes:
+        if isinstance(shape, int) and shape < 0:
+            raise RuntimeError(rf"Trying to create tensor with negative dimension {shape}: [{shape}]")
+        if isinstance(shape, tuple):
+            for ele in shape:
+                if ele < 0:
+                    raise RuntimeError(rf"Trying to create tensor with negative dimension {ele}: [{ele}]")
+
+    try:
+        result = [1] * max(map(len, shapes))
```

Comment:
don't rely on `try/except` because it is typically slow, rather check `isinstance`. Also, this won't work because you can send valid inputs where neither all are iterables (and thus `len` can be called on them) nor integers (and thus `max` can be called on them):
```
In [4]: torch.broadcast_shapes(1,(5,))
Out[4]: torch.Size([5])

In [5]: shapes=(1,(5,))

In [6]: max(map(len,shapes))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-2bcf66de43a2> in <module>
----> 1 max(map(len,shapes))

TypeError: object of type 'int' has no len()

In [7]: max(shapes)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-d00078ec8f13> in <module>
----> 1 max(shapes)

TypeError: '>' not supported between instances of 'tuple' and 'int'

```

Reply:
Omg, I'd missed this case earlier. Thanks for the info. I've made the amendments. Thanks!